tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -9,147 +9,155 @@
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(LiveDebugger));
+        private static readonly object GlobalLock = new();
 
         private readonly ImmutableDebuggerSettings _settings;
         private readonly IDiscoveryService _discoveryService;
-        private readonly IConfigurationPoller _configurationPoller;
+        private readonly IRemoteConfigurationManager _remoteConfigurationManager;
         private readonly IDebuggerSink _debuggerSink;
         private readonly ILineProbeResolver _lineProbeResolver;
         private readonly List<ProbeDefinition> _unboundProbes;
         private readonly IProbeStatusPoller _probeStatusPoller;
-        private readonly object _locker = new();
-
-        private LiveDebugger(ImmutableDebuggerSettings settings, IDiscoveryService discoveryService, IConfigurationPoller configurationPoller, ILineProbeResolver lineProbeResolver, IDebuggerSink debuggerSink, IProbeStatusPoller probeStatusPoller)
+        private readonly ConfigurationUpdater _configurationUpdater;
+        private readonly object _instanceLock = new();
+        private bool _isInitialized;
+
+        private LiveDebugger(
+            ImmutableDebuggerSettings settings,
+            string serviceName,
+            IDiscoveryService discoveryService,
+            IRemoteConfigurationManager remoteConfigurationManager,
+            ILineProbeResolver lineProbeResolver,
+            IDebuggerSink debuggerSink,
+            IProbeStatusPoller probeStatusPoller,
+            ConfigurationUpdater configurationUpdater)
         {
             _settings = settings;
             _discoveryService = discoveryService;
-            _configurationPoller = configurationPoller;
             _lineProbeResolver = lineProbeResolver;
             _debuggerSink = debuggerSink;
             _probeStatusPoller = probeStatusPoller;
+            _remoteConfigurationManager = remoteConfigurationManager;
+            _configurationUpdater = configurationUpdater;
             _unboundProbes = new List<ProbeDefinition>();
+            Product = new LiveDebuggerProduct(serviceName);
+            ServiceName = serviceName;
         }
 
-        public static LiveDebugger Instance => LazyInstance.Value;
+        public static LiveDebugger Instance { get; private set; }
+
+        public LiveDebuggerProduct Product { get; }
+
+        public string ServiceName { get; }
 
-        private static LiveDebugger Create()
+        public static LiveDebugger Create(
+            ImmutableDebuggerSettings settings,
+            string serviceName,
+            IDiscoveryService discoveryService,
+            IRemoteConfigurationManager remoteConfigurationManager,
+            ILineProbeResolver lineProbeResolver,
+            IDebuggerSink debuggerSink,
+            IProbeStatusPoller probeStatusPoller,
+            ConfigurationUpdater configurationUpdater)
```

Comment:
I'm not sure I agree.

For me `IDiscoveryService` is some kind of settings, which is resolved by sending requests to agent, while `IRemoteConfigurationManager` handling all `product` stuff.

I think removing dependency of `IRemoteConfigurationManager` is making more sense, since we can (and probably should) register product outside of `LiveDebugger` initialization process.
But this will require small refactor of the `DebuggerEnabled` setting, we don't want to register live debugger, if it not enabled.

WDYT?

Reply:
Hmmm, I guess 🤔 I guess I'm getting confused about the delineation between discovery service and RCM... DiscoveryService is about "does the agent support this product/feature" (which _could_ change throughout the lifetime of the app), and RCM is about _configuration_ for this product/feature has changed. 

It feels like those are two halves of the same thing to me, and that we could combine them into one, but just something to consider for a future PR I think

Reply:
Well, yes, agree with you, with some small clarification, that not every product supports remote configuration.
So, discovery service can be used separately from remote configuration.

I think we need more usages of those things, to be sure, if we may combine them into one.

Reply:
Totally agree that this would be great to address in a future PR. Some thoughts for that potential future refactoring;:

> DiscoveryService is about "does the agent support this product/feature" (which could change throughout the lifetime of the app)

Exactly, perhaps to make this clearer, we should rename it to the name that is used for the equivalent class in the Java tracer, IIRC `DDAgentFeatureDiscoveryService` 
 
> RCM is about configuration for this product/feature has changed.

That's mostly true, but there is also the case of the new "1-click product/feature activation" idea that is currently being introduced for ASM but will hopefully be used broadly across all of our products in the future. Since the question of whether a particular product/feature *should* and *can* be enabled depends on both these things, and both could change during the lifetime of the app, I tend to think it would be a good idea to introduce a new class that is responsible for handling feature/product activation in a consistent way, and encapsulates both 
`DDAgentFeatureDiscoveryService` and `IRemoteConfigManager` for this purpose. /CC @robertpi @daniel-romano-DD